automatic/youtube-dl/update.ps1

```diff
@@ -1,20 +1,36 @@
 import-module au
+import-module "$env:ChocolateyInstall\helpers\chocolateyInstaller.psm1"
 
 $domain   = 'https://github.com'
 $releases = "$domain/rg3/youtube-dl/releases/latest"
 
-function global:au_afterUpdate {
-  rm $PSScriptRoot/tools/*.exe -ea Ignore
+function global:au_BeforeUpdate {
+  if (Test-Path "$PSScriptRoot\tools") {
+    Remove-Item "$PSScriptRoot\tools\*.exe" -Force
+  } else {
+    New-Item -ItemType Directory "$PSScriptRoot\tools"
+  }
+  $Latest.FileName = Get-WebFileName $Latest.URL32 'youtube-dl.exe'
+  $filePath = "$PSScriptRoot\tools\$($Latest.FileName)"
+  Get-WebFile $Latest.URL32 $filePath
+
+  # Let us check if the checksum actually matches
+  $actualChecksum = Get-FileHash -Algorithm $Latest.ChecksumType32 $filePath | % Hash
+
+  if ($actualChecksum -ne $Latest.Checksum32) {
```

Comment:
the software provided checksum is aquired here:
https://github.com/chocolatey/chocolatey-coreteampackages/pull/513/files#diff-9fff76ad80fcbf680f7450c0663434d4R44

since there wasn't anything else, I'll merge this now and push out the fix